src/pretix/base/services/cart.py

```diff
@@ -1611,13 +1610,14 @@ def clear_cart(self, event: Event, cart_id: str=None, locale='en', sales_channel
 
 
 @app.task(base=ProfiledEventTask, bind=True, max_retries=5, default_retry_delay=1, throws=(CartError,))
-def set_cart_addons(self, event: Event, addons: List[dict], cart_id: str=None, locale='en',
+def set_cart_addons(self, event: Event, addons: List[dict], add_to_cart_items: List[dict], cart_id: str=None, locale='en',
                     invoice_address: int=None, sales_channel='web', override_now_dt: datetime=None) -> None:
     """
-    Removes a list of items from a user's cart.
+    Assigns addons to eligible products in a user's cart, adding and removing the addon products as necessary to
+    ensure the requested addon state.
     :param event: The event ID in question
     :param addons: A list of dicts with the keys addon_to, item, variation
```

Comment:
Should we document the `add_to_cart_items` parameter here as well? Or is it kind of self-explanatory like locale, invoice_adress, etc. Addons is documented, so I guess we should add it.

Reply:
```suggestion
    :param addons: A list of dicts with the keys addon_to, item, variation
    :param add_to_cart_items: A list of dicts with the keys item, variation, count, custom_price, voucher, seat ID
```